openmdao/drivers/genetic_algorithm_driver.py

```diff
@@ -200,6 +204,33 @@ def _setup_comm(self, comm):
         self._concurrent_color = 0
         return comm
 
+    def _setup_recording(self):
+        """
+        Set up case recording.
+        """
+        if MPI:
+            run_parallel = self.options['run_parallel']
+            procs_per_model = self.options['procs_per_model']
+
+            for recorder in self._rec_mgr:
+                if run_parallel:
+                    # write cases only on procs up to the number of parallel DOEs
+                    # (i.e. on the root procs for the cases)
+                    if procs_per_model == 1:
+                        recorder.record_on_proc = True
+                    else:
+                        size = self._problem_comm.size // procs_per_model
+                        if self._problem_comm.rank < size:
+                            recorder.record_on_proc = True
+                        else:
+                            recorder.record_on_proc = False
+
+                elif self._problem_comm.rank > 0:
+                    # if not running cases in parallel, then just record on proc 0
+                    recorder.record_on_proc = False
```

Comment:
So GE is telling their recorder "Don't record on on procs 1-N, but do whatever your default behavior is on rank 0." (and current default behavior value is `_Undefined`, which behaves as True.) I feel like there is a slight disconnect here. I think the driver should either 1.) explicitly tell the recorder to record on 0 and not to record on 1-N, or 2.) just let openmdao pick one single rank to record on, since it is mostly irrelevant to the driver when run_parallel is False.  What we have here is a hybrid and doesn't feel consistent.

Reply:
The idea was that if you don't touch the `record_on_proc` flag then you get the default behavior, which is to only record on proc 0.  If you set `record_on_proc` to True on any procs then those procs and only those procs will record.  

I guess setting the flag to False is pointless.. since we only care if the flag is True or not.

I tried to express this in line 82 of case_recorder.py..   looks like I should embellish the explanation.

Reply:
reworked the logic..  no longer set False, since it only matters if it's True..  updated explanations in CaseRecorder